test/e2e/kubectl/kubectl.go

```diff
@@ -966,31 +973,13 @@ metadata:
 			ginkgo.By("sleep for 10s to wait for potential crd openapi publishing alpha feature")
 			time.Sleep(10 * time.Second)
 
-			publishedSchema := schemaForGVK(schema.GroupVersionKind{Group: crd.Crd.Spec.Group, Version: crd.Crd.Spec.Version, Kind: crd.Crd.Spec.Names.Kind})
-			expectSuccess := false
-			if publishedSchema == nil || publishedSchema.Properties == nil || publishedSchema.Properties.AdditionalProperties == nil || len(publishedSchema.Properties.AdditionalProperties) == 0 {
-				// expect success in the following cases:
-				// - no schema was published
-				// - a schema was published with no properties
-				expectSuccess = true
-				e2elog.Logf("no schema with properties found, expect apply with extra properties to succeed")
-			} else {
-				e2elog.Logf("schema with properties found, expect apply with extra properties to fail")
-			}
+			schemaForGVK(schema.GroupVersionKind{Group: crd.Crd.Spec.Group, Version: crd.Crd.Spec.Versions[0].Name, Kind: crd.Crd.Spec.Names.Kind})
```

Comment:
Is there a reason we're calling this now? We're not using the result… I could see requiring a definition be returned (failing on nil)

Reply:
Good catch. Added the nil check.